server/src/main.rs

```diff
@@ -344,114 +312,93 @@ impl Backend {
         )
     }
 
-    async fn get_sources_directory(
+    async fn get_reference_directories(
```

Comment:
How is this function and its logic different from the various functions on `SliceConfig`.